cmd/pulumi-test-language/providers/config_grpc_provider.go

```diff
@@ -0,0 +1,331 @@
+			InputProperties: configSpec.Variables,
+			ObjectTypeSpec: pschema.ObjectTypeSpec{
+				Properties: configSpec.Variables,
+			},
+		},
+		Types: types,
+		Resources: map[string]pschema.ResourceSpec{
+			fmt.Sprintf("%s:index:ConfigGetter", p.Pkg()): {
+				ObjectTypeSpec: pschema.ObjectTypeSpec{
+					Properties: map[string]pschema.PropertySpec{
+						"config": {
+							TypeSpec: pschema.TypeSpec{Type: "string"},
+						},
+					},
+					Required: []string{"config"},
+				},
+			},
+		},
+		Functions: map[string]pschema.FunctionSpec{},
+		Language: map[string]pschema.RawMessage{
+			"nodejs": []byte(`{"respectSchemaVersion": true}`),
+		},
+	}
+
+	toSecretSchema := p.generateSchema(types, 0, 0)
+	allProps := []string{}
+	for k := range toSecretSchema {
+		allProps = append(allProps, k)
+	}
+
+	schema.Functions[fmt.Sprintf("%s:index:toSecret", p.Pkg())] = pschema.FunctionSpec{
+		Inputs: &pschema.ObjectTypeSpec{
+			Properties: toSecretSchema,
+		},
+		Outputs: &pschema.ObjectTypeSpec{
+			Properties: toSecretSchema,
+			Required:   allProps,
+		},
+	}
+
+	return schema
+}
+
+func (p *ConfigGrpcProvider) NewProviderServer() pulumirpc.ResourceProviderServer {
+	return &configGrpcProviderServer{
+		ResourceProviderServer: plugin.NewProviderServer(p),
+	}
+}
+
+func (p *ConfigGrpcProvider) GetSchema(
+	context.Context, plugin.GetSchemaRequest,
+) (plugin.GetSchemaResponse, error) {
+	schema := p.schema()
+	schemaBytes, err := json.Marshal(schema)
+	if err != nil {
+		return plugin.GetSchemaResponse{}, err
+	}
+	return plugin.GetSchemaResponse{Schema: schemaBytes}, nil
+}
+
+func (p *ConfigGrpcProvider) GetPluginInfo(context.Context) (workspace.PluginInfo, error) {
+	ver := semver.MustParse(p.version())
+	return workspace.PluginInfo{
+		Version: &ver,
+	}, nil
+}
+
+func (p *ConfigGrpcProvider) Check(
+	_ context.Context, req plugin.CheckRequest,
+) (plugin.CheckResponse, error) {
+	return plugin.CheckResponse{Properties: req.News}, nil
+}
+
+func (p *ConfigGrpcProvider) CheckConfig(
+	_ context.Context, req plugin.CheckConfigRequest,
+) (plugin.CheckConfigResponse, error) {
+	return plugin.CheckConfigResponse{Properties: req.News}, nil
+}
+
+func (p *ConfigGrpcProvider) Configure(
+	context.Context, plugin.ConfigureRequest,
+) (plugin.ConfigureResponse, error) {
+	return plugin.ConfigureResponse{}, nil
+}
+
+func (p *ConfigGrpcProvider) DiffConfig(
+	context.Context, plugin.DiffConfigRequest,
+) (plugin.DiffConfigResponse, error) {
+	return plugin.DiffResult{}, nil
+}
+
+func (p *ConfigGrpcProvider) Invoke(
+	ctx context.Context, req plugin.InvokeRequest,
+) (plugin.InvokeResponse, error) {
+	switch {
+	case string(req.Tok) == fmt.Sprintf("%s:index:toSecret", p.Pkg()):
+		secreted := req.Args.Copy()
+		for k, v := range secreted {
+			secreted[k] = resource.MakeSecret(v)
+		}
+		return plugin.InvokeResponse{Properties: secreted}, nil
+	default:
+		return plugin.InvokeResponse{}, errors.New("Unknown function")
+	}
+}
```

Comment:
Might be worth removing this for now if the invoke is not currently being used.

Reply:
The toSecret invoke is used to inject first class secrets. I couldn't find a native way to do it though there might be one actually.